doc/source/data/examples/batch_inference_object_detection.ipynb

```diff
@@ -0,0 +1,350 @@
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(4,) (478, 612, 3)\n"
+     ]
+    }
+   ],
+   "source": [
+    "single_batch = ds.take_batch(batch_size=4)\n",
+    "print(single_batch[\"image\"].shape, single_batch[\"image\"][0].shape)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Batch inference with Ray Data\n",
+    "\n",
+    "Then we will demostrate how to use Ray Data to do batch inference in a distributed cluster.\n",
+    "\n",
+    "The first step is to package the model in a Python class. The class is mainly consist of two parts. In the `__init__` constructor, we package the code that loads and initializes the model. And in the `__call__` method, we package the code that applies model on each batch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import Dict\n",
+    "\n",
+    "import numpy as np\n",
+    "from torchvision import transforms\n",
+    "from torchvision.models.detection import (FasterRCNN_ResNet50_FPN_V2_Weights,\n",
+    "                                          fasterrcnn_resnet50_fpn_v2)\n",
+    "\n",
+    "\n",
+    "class ObjectDetectionModel:\n",
+    "    def __init__(self):\n",
+    "        # Define the model loading and initialization code in `__init__`\n",
+    "        self.weights = FasterRCNN_ResNet50_FPN_V2_Weights.DEFAULT\n",
+    "        self.model = fasterrcnn_resnet50_fpn_v2(\n",
+    "            weights=self.weights,\n",
+    "            box_score_thresh=0.9,\n",
+    "        )\n",
+    "        self.model.eval()\n",
+    "        # Note, since the data in the batch input is `np.ndarray`s, \n",
+    "        # we need `transforms.ToTensor` to convert the data to torch tensors.\n",
+    "        self.preprossor = transforms.Compose(\n",
+    "            [transforms.ToTensor(), self.weights.transforms()]\n",
+    "        )\n",
+    "\n",
+    "    def __call__(self, batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:\n",
+    "        # Define the per-batch inference code in `__call__`\n",
+    "        # Preprocess the images.\n",
+    "        batch = [self.preprossor(image) for image in batch[\"image\"]]\n",
+    "        # Do inference on the images.\n",
+    "        predictions = self.model(batch)\n",
+    "        # Get the inferred labels and convert it to a np.ndarray\n",
+    "        labels = np.array(\n",
+    "            [\n",
+    "                \",\".join(\n",
+    "                    [self.weights.meta[\"categories\"][i] for i in prediction[\"labels\"]]\n",
+    "                )\n",
+    "                for prediction in predictions\n",
+    "            ],\n",
+    "            dtype=\"S\",\n",
+    "        )\n",
```

Comment:
for object detection, we need the actual bounding boxes for each category, not just the labels right?

See https://github.com/ray-project/ray/blob/fb1394e90a4a6bb11a0f315061a3230ab4ab4c2f/doc/source/ray-air/examples/pytorch_object_detection.ipynb as an example.